### DISCUSSION TITLE
Discuss about Stochastic duration predictor and adversarial training

### DISCUSSION THREAD

**Discussion by ductho9799** [p0p4k/vits2_pytorch#34] [General]
In Section 2.1 of VITS-2, I see "Our proposed duration predictor and training mechanism allow for a learning duration in **short steps**, and the duration predictor is **separately** trained as the last training step, which reduces the overall computation time for training."

So, I think the duration predictor network is seperately. When training Duration Predictor as the last training step, and the others modules such as: TextEncoder, Posterior Encoder, Generator, Normalizing Flow are freeze.

**Comment by p0p4k** [comment]
Yes, you are right. See this thread for more  [discussion](https://github.com/p0p4k/vits2_pytorch/issues/3#issuecomment-1671752391). I will add it later on or wait for someone to send a PR =)
